### PR TITLE
Use response selector for FAQs

### PR DESCRIPTION
**Pull request by melindaloubser1**
* Use response selector for FAQs
* Handle follow up "explain" faqs in custom action
closes https://github.com/RasaHQ/rasa-demo/issues/344
Outstanding issue:
Does markdown render correctly in channel, if not rasa x?

### CODE REVIEW

**Comment by melindaloubser1** [comment]
@akelad re. URLs in response selector text - actually improves results. 
## Results with URLs
```  
  "accuracy": 0.9636194029850746,
  "macro avg": {
    "precision": 0.8645443196004994,
    "recall": 0.8888888888888888,
    "f1-score": 0.8733013589128696,
    "support": 1072
  },
  "weighted avg": {
    "precision": 0.9431808653362401,
    "recall": 0.9636194029850746,
    "f1-score": 0.9505328572962526,
    "support": 1072
  }
```
## Results without URLs:
```
  "accuracy": 0.9048507462686567,
  "macro avg": {
    "precision": 0.7370103916866506,
    "recall": 0.7777777777777778,
    "f1-score": 0.7455808080808081,
    "support": 1072
  },
  "weighted avg": {
    "precision": 0.8795232470739826,
    "recall": 0.9048507462686567,
    "f1-score": 0.8848477781546812,
    "support": 1072
  }
```

**Comment by melindaloubser1** [comment]
Since response selector is trained on the text of the response itself, how do we improve the performance issues for the 3 badly-performing faqs?

**Comment by akelad** [comment]
what do rasax and differencerasarasax get confused with?

**Comment by melindaloubser1** [comment]
Since it doesn't report that, I can only see anecdotally that both are getting confused with `channels`

**Comment by akelad** [comment]
@dakshvar22 any ideas on how we can figure that out/other things to try?

**Comment by dakshvar22** [comment]
If you use the latest `1.9.x` branch, the report should mention what these two faqs are getting confused with just like intents. Can you try that once?

**Comment by melindaloubser1** [comment]
@daksh I'm not seeing confused with, but after retraining on latest 1.9.x, results are 1.0 everywhere - what else changed in response selector?

**Comment by melindaloubser1** [comment]
report looks like this:
```
  "Follow along with the [Rasa Masterclass]  to build a level 3 assistant from scratch, or check out our [blog]  for in-depth tutorials on customizing your assistant.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 37
  },
  "Rasa has more than 7000 community members and over 300 code and content contributors 🚀.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 40
  },
  "accuracy": 1.0,
  "macro avg": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 1072
  },
  "weighted avg": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 1072
  }
}
```

**Comment by dakshvar22** [comment]
i am sorry, i just checked - `confused_with` was merged into master.

**Comment by melindaloubser1** [comment]
Ah ok, I can go try that too, but curious about change in performance in any case.

**Comment by dakshvar22** [comment]
There were two bugfixes that we merged post 1.9.0 :
https://github.com/RasaHQ/rasa/pull/5505
https://github.com/RasaHQ/rasa/pull/5626

**Comment by dakshvar22** [comment]
But are you testing on the train set itself? Accuracy of 1 across the board seems fishy

**Comment by melindaloubser1** [comment]
Yes, this was on the train set, so that's why I was concerned when it wasn't 1 before. Haven't tried it on a test set yet.

**Comment by melindaloubser1** [comment]
Using master branch, on train set, with URLs in responses:
```json
{
  "Rasa has more than 7000 community members and over 300 code and content contributors 🚀.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 40,
    "confused_with": {}
  },
  "Rasa Masterclass has a great introduction to dialogue management [here](https://www.youtube.com/watch?v=WoTmnN2EMdo).": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 14,
    "confused_with": {
      "Slots act as a key-value store which can be used to store information the user provided (e.g their home city) as well as information gathered about the outside world (e.g. the result of a database query). Learn more [here](https://rasa.com/docs/rasa/core/slots/).": 14
    }
  },
  "Rasa X is a toolset that helps you leverage conversations to improve your assistant. Learn more about Rasa X [here](https://rasa.com/docs/rasa-x/) or read the blog post about why we developed it [here](https://blog.rasa.com/algorithms-alone-wont-solve-conversational-ai-introducing-rasa-x/).": {
    "precision": 0.5,
    "recall": 1.0,
    "f1-score": 0.6666666666666666,
    "support": 23,
    "confused_with": {}
  },
  "The supported Python versions are 3.6 and 3.7 🐍. The recommended version is 3.7.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 14,
    "confused_with": {}
  },
  "We have a [comprehensive list of supported connectors](https://rasa.com/docs/core/connectors/), but if you don't see the one you're looking for, you can always create a custom connector by following [this guide](https://rasa.com/docs/rasa/user-guide/connectors/custom-connectors/).": {
    "precision": 0.49019607843137253,
    "recall": 1.0,
    "f1-score": 0.6578947368421052,
    "support": 50,
    "confused_with": {}
  },
  "NLU’s job is to interpret messages, and Core’s job is to decide what should happen next.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 59,
    "confused_with": {}
  },
  "Rasa Open Source is written in Python, but programming is not required to develop an assistant. To learn how to create an assistant, check out the [Rasa Masterclass](https://www.youtube.com/watch?v=rlAQWbhwqLA). There is one place where programming skills come in handy, and that is when implementing [custom actions](https://rasa.com/docs/rasa/core/actions/#custom-actions). You can create an action server in Node.js, .NET, Java, or any other language and define your actions there. For custom actions written in Python, we provide a small [Python SDK](https://rasa.com/docs/rasa/api/rasa-sdk/#rasa-sdk) to make development even easier.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 49,
    "confused_with": {}
  },
  "Slots act as a key-value store which can be used to store information the user provided (e.g their home city) as well as information gathered about the outside world (e.g. the result of a database query). Learn more [here](https://rasa.com/docs/rasa/core/slots/).": {
    "precision": 0.8157894736842105,
    "recall": 1.0,
    "f1-score": 0.898550724637681,
    "support": 62,
    "confused_with": {}
  },
  "Rasa Masterclass episodes 2-4 focus on NLU. Check out episode 2 [here](https://www.youtube.com/watch?v=k5UeywXA28k).": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 15,
    "confused_with": {}
  },
  "Follow along with the [Rasa Masterclass](https://www.youtube.com/watch?v=rlAQWbhwqLA) to build a level 3 assistant from scratch, or check out our [blog](https://blog.rasa.com/tag/tutorials/) for in-depth tutorials on customizing your assistant.": {
    "precision": 0.8222222222222222,
    "recall": 1.0,
    "f1-score": 0.9024390243902439,
    "support": 37,
    "confused_with": {}
  },
  "You can connect Rasa to any channel you like, including voice channels. You can find more information [here](https://rasa.com/docs/core/connectors/).": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 72,
    "confused_with": {}
  },
  "Rasa Open Source is a framework you can use to build an assistant. Rasa X is a toolset built on top of Rasa Open Source that comes with a UI that helps you learn from real conversations and improve your assistant. Rasa X is completely optional --if you'd like, you can use Rasa Open Source on its own.": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 16,
    "confused_with": {
      "Follow along with the [Rasa Masterclass](https://www.youtube.com/watch?v=rlAQWbhwqLA) to build a level 3 assistant from scratch, or check out our [blog](https://blog.rasa.com/tag/tutorials/) for in-depth tutorials on customizing your assistant.": 8,
      "Rasa Enterprise is an enterprise-ready subscription to develop and ship contextual assistants at scale, which comes features for increased security and privacy, as well as expert support.": 5,
      "We have a [comprehensive list of supported connectors](https://rasa.com/docs/core/connectors/), but if you don't see the one you're looking for, you can always create a custom connector by following [this guide](https://rasa.com/docs/rasa/user-guide/connectors/custom-connectors/).": 2
    }
  },
  "I can't help you with that, I'm sorry.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 333,
    "confused_with": {}
  },
  "You can use Rasa to build assistants in any language you want!": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 93,
    "confused_with": {}
  },
  "\"[Rasa Forum](https://forum.rasa.com) is the place where all the community buzz happens. It's a place to join discussions, get the answers to your technical questions and get the latest updates on what's happening at Rasa.\"": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 50,
    "confused_with": {
      "We have a [comprehensive list of supported connectors](https://rasa.com/docs/core/connectors/), but if you don't see the one you're looking for, you can always create a custom connector by following [this guide](https://rasa.com/docs/rasa/user-guide/connectors/custom-connectors/).": 50
    }
  },
  "Rasa Enterprise is an enterprise-ready subscription to develop and ship contextual assistants at scale, which comes features for increased security and privacy, as well as expert support.": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 22,
    "confused_with": {
      "Rasa X is a toolset that helps you leverage conversations to improve your assistant. Learn more about Rasa X [here](https://rasa.com/docs/rasa-x/) or read the blog post about why we developed it [here](https://blog.rasa.com/algorithms-alone-wont-solve-conversational-ai-introducing-rasa-x/).": 22
    }
  },
  "I only understand English, I'm sorry.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 21,
    "confused_with": {}
  },
  "\"The Rasa machine learning framework is open source! We strive to provide well-maintained and documented open source tools, allowing all makers to build the best AI assistants without friction.\\nIn addition to that, we offer the free toolset [Rasa X](https://rasa.com/docs/rasa-x/) and paid Rasa Enterprise subscriptions.\\n\\nYou can find out more [here](https://rasa.com/product/pricing/) about the plans and pricing.\"": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 102,
    "confused_with": {}
  },
  "accuracy": 0.9048507462686567,
  "macro avg": {
    "precision": 0.7015670985743225,
    "recall": 0.7777777777777778,
    "f1-score": 0.7291972862520386,
    "support": 1072
  },
  "weighted avg": {
    "precision": 0.8535550126046753,
    "recall": 0.9048507462686567,
    "f1-score": 0.8725078908632592,
    "support": 1072
  }
}
```

**Comment by dakshvar22** [comment]
Can you try `scale_loss:False` for the `faq` ResponseSelector. I get 0.97 as the test micro average f1 with that.

**Comment by melindaloubser1** [comment]
@dakshvar22 On the train set still getting bad results (with `scale_loss=False`)
```
  "macro avg": {
    "precision": 0.8888888888888888,
    "recall": 0.8888888888888888,
    "f1-score": 0.8888888888888888,
    "support": 1072
  },
  "weighted avg": {
    "precision": 0.6697761194029851,
    "recall": 0.6697761194029851,
    "f1-score": 0.6697761194029851,
    "support": 1072
  }
```

**Comment by dakshvar22** [comment]
That is surprising. I created a train test split from the data on this branch using `rasa data split nlu` and trained locally on my side using the current rasa master branch. 

Evaluating on training set - 
```
"accuracy": 0.9824970828471412,
  "macro avg": {
    "precision": 0.9139846743295019,
    "recall": 0.9440251572327044,
    "f1-score": 0.9252606474097702,
    "support": 857
  },
  "weighted avg": {
    "precision": 0.9743612441153986,
    "recall": 0.9824970828471412,
    "f1-score": 0.9771605851551193,
    "support": 857
  }
```

Evaluation on test set - 
```
"accuracy": 0.9767441860465116,
  "macro avg": {
    "precision": 0.840936840936841,
    "recall": 0.8842592592592592,
    "f1-score": 0.857036526985376,
    "support": 215
  },
  "weighted avg": {
    "precision": 0.9629997909067676,
    "recall": 0.9767441860465116,
    "f1-score": 0.9688055488198235,
    "support": 215
  }
```

**Comment by melindaloubser1** [comment]
If you train and test without splitting, do you see the same results I do? Otherwise there must be something wrong with my environment where I installed master.

**Comment by dakshvar22** [comment]
I think I know what's going wrong with your setup. To save time are you training only the faq response selector and have commented out the out_of_scope response selector in your config?
While testing, responses for the out_of_scope messages would be compared against an empty string.
I made the same mistake and then included the out_of_scope response selector too, and got f1 of 1 across the board.

**Comment by melindaloubser1** [comment]
Oh! You're right, that's what I was doing. I'll try taking out the `out_of_scope` data and see what that gives.

What can we do about the underperforming one?

**Comment by dakshvar22** [comment]
You don't need to take out the `out_of_scope` data, just include the response selector for `out_of_scope` one as well in your config.

**Comment by melindaloubser1** [comment]
I figured both ways round would work

**Comment by melindaloubser1** [comment]
Yup, all good then - 1.0 across the board (all data train & test). I'm trying the same thing with 1.9.5 now

**Comment by dakshvar22** [comment]
`1.9.6` I guess? That's the latest

**Comment by melindaloubser1** [comment]
Right, my bad

**Comment by melindaloubser1** [comment]
Yup, all good then.

**Comment by melindaloubser1** [comment]
Thanks for the help @dakshvar22 . @rgstephens  this is ready for review =)

**Comment by akelad** [comment]
@erohmensing there's another weird black behaviour here

File: actions/actions.py [hunk 1]
```diff
@@ -113,10 +113,7 @@ def slot_mappings(self) -> Dict[Text, Union[Dict, List[Dict]]]:
                 self.from_entity(entity="job_function"),
                 self.from_text(intent="enter_data"),
             ],
-            "use_case": [
-                self.from_text(intent="enter_data"),
-                self.from_intent(intent="ask_faq_voice", value="voice"),
-            ],
+            "use_case": [self.from_text(intent="enter_data"),],
```
Comment (akelad) [inline actions/actions.py:116]: are we sure about this? I think people can say "a voice bot" in response to the use case

Comment (akelad) [reply actions/actions.py:116]: ohhh yes true. damn it

Comment (melindaloubser1) [reply actions/actions.py:116]: retrieval actions are kind a double edged sword

**Comment by akelad** [comment]
@melindaloubser1 I think we can merge this now right (after resolving conflicts ofc)?

**Comment by melindaloubser1** [comment]
Question - how can I check that the responses render correctly in the website chat channel?

**Comment by erohmensing** [comment]
@melindaloubser1 for rendering, you can copy the widget code into your own index.html and run the bot locally

**Comment by melindaloubser1** [comment]
@erohmensing hyperlinks and emojis render fine, newlines don't.   For most of the responses we can just remove the newlines and have it all in one text block, but this one would obv. look funny:
image

Could we just change it to a non list? ` You can ask me about how to get started with Rasa, the difference between Rasa and Rasa  X, subscribing to our newsletter or booking a sales call.`

**Comment by erohmensing** [comment]
Yeah, makes the most sense for now. Do we have a list of drawbacks we've encountered with the response selector? I feel like we've noticed there are quite a few tradeoffs, would be good to collect them

**Comment by akelad** [comment]
ugh that's actually extremely annoying that that doesn't render

**Comment by melindaloubser1** [comment]
I added some things about response selector to the confusing story stuff doc. 
The newline thing is already a to do: https://github.com/RasaHQ/rasa/issues/5521

**Comment by melindaloubser1** [comment]
All the `/chitchat` responses get scores of 0.0 (on train set), but seem to be returned just fine in practice.  I think it's fine to merge since it's working but still odd @dakshvar22
```
{
  "Rasa Masterclass has a great introduction to dialogue management [here](https://www.youtube.com/watch?v=WoTmnN2EMdo).": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 14
  },
  "NLU’s job is to interpret messages, and Core’s job is to decide what should happen next.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 60
  },
  " Likewise!": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 47
  },
  " Yep, I'm a bot!": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 37
  },
  "You can use Rasa to build assistants in any language you want!": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 94
  },
  "Slots act as a key-value store which can be used to store information the user provided (e.g their home city) as well as information gathered about the outside world (e.g. the result of a database query). Learn more [here](https://rasa.com/docs/rasa/core/slots/).": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 62
  },
  "The Rasa machine learning framework is open source! We strive to provide well-maintained and documented open source tools, allowing all makers to build the best AI assistants without friction. In addition to that,  we offer the free toolset [Rasa X](https://rasa.com/docs/rasa-x/) and paid Rasa Enterprise   subscriptions. You can find out more [here](https://rasa.com/product/pricing/)  about the plans and pricing.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 104
  },
  "The supported Python versions are 3.6 and 3.7 🐍. The recommended version is 3.7.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 14
  },
  "You can connect Rasa to any channel you like, including voice channels. You can find more information [here](https://rasa.com/docs/core/connectors/).": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 72
  },
  " I hope you are being yourself.": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 26
  },
  " I'm sorry, I can't recommend you a restaurant as I usually cook at home.": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 55
  },
  " That's not very nice 😢": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 99
  },
  "Follow along with the [Rasa Masterclass](https://www.youtube.com/watch?v=rlAQWbhwqLA) to build a level 3 assistant from scratch, or check out our [blog](https://blog.rasa.com/tag/tutorials/) for in-depth tutorials on customizing your assistant.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 37
  },
  " It's the most wonderful time of the year!": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 51
  },
  "Rasa Open Source is written in Python, but programming is not required to develop an assistant. To learn how to create an assistant, check out the [Rasa Masterclass](https://www.youtube.com/watch?v=rlAQWbhwqLA). There is one place where programming skills come in handy, and that is when implementing [custom actions](https://rasa.com/docs/rasa/core/actions/#custom-actions). You can create an action server in Node.js, .NET, Java, or any other language and define your actions there. For custom actions written in Python, we provide a small [Python SDK](https://rasa.com/docs/rasa/api/rasa-sdk/#rasa-sdk) to make development even easier.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 49
  },
  "I only understand English, I'm sorry.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 26
  },
  " I'm great! Thanks for asking.": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 90
  },
  "Rasa Enterprise is an enterprise-ready subscription to develop and ship contextual assistants at scale, which comes features for increased security and privacy, as well as expert support.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 22
  },
  " Rasa Open Source is a conversational AI framework for that provides machine learning tools for building contextual assistants. You can read more about it here in the docs or on our [website](https://rasa.com/).": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 75
  },
  " You can ask me about how to get started with Rasa, the difference between Rasa and Rasa X, subscribing to our newsletter or booking a sales call.": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 101
  },
  "We have a [comprehensive list of supported connectors](https://rasa.com/docs/core/connectors/), but if you don't see the one you're looking for, you can always create a custom connector by following [this guide](https://rasa.com/docs/rasa/user-guide/connectors/custom-connectors/).": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 50
  },
  " I don't know about where you live, but in my world it's always sunny": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 78
  },
  " Why did the robot cross the road? -Because he was programmed to.": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 40
  },
  " I was built with a lot of love and patience.": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 46
  },
  " I was born in Berlin, but I consider myself a citizen of the world.": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 49
  },
  " Old enough to be a bot!": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 53
  },
  " I can spell baguette in French, but unfortunately English is the only language I can answer you in.": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 57
  },
  " It's probably the one that your parents chose for you.": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 37
  },
  "Rasa X is a toolset that helps you leverage conversations to improve your assistant. Learn more about Rasa X [here](https://rasa.com/docs/rasa-x/) or read the blog post about why we developed it [here](https://blog.rasa.com/algorithms-alone-wont-solve-conversational-ai-introducing-rasa-x/).": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 23
  },
  "[Rasa Forum](https://forum.rasa.com) is the place where all the community buzz happens. It's a place to join discussions, get the answers to your technical questions and get the latest updates on what's happening at Rasa.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 50
  },
  "Rasa has more than 7000 community members and over 300 code and content contributors 🚀.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 40
  },
  " I'm Sara, the Rasa bot! At the same time I'm also the Rasa mascot 🐦": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 62
  },
  "I can't help you with that, I'm sorry.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 341
  },
  " The engineers at Rasa 🛠": {
    "precision": 0.0,
    "recall": 0.0,
    "f1-score": 0.0,
    "support": 105
  },
  "Rasa Masterclass episodes 2-4 focus on NLU. Check out episode 2 [here](https://www.youtube.com/watch?v=k5UeywXA28k).": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 15
  },
  "Rasa Open Source is a framework you can use to build an assistant. Rasa X is a toolset built on top of Rasa Open Source that comes with a UI that helps you learn from real conversations and improve your assistant. Rasa X is completely optional --if you'd like, you can use Rasa Open Source on its own.": {
    "precision": 1.0,
    "recall": 1.0,
    "f1-score": 1.0,
    "support": 18
  },
  "micro avg": {
    "precision": 1.0,
    "recall": 0.4961346066393815,
    "f1-score": 0.6632218844984803,
    "support": 2199
  },
  "macro avg": {
    "precision": 0.5,
    "recall": 0.5,
    "f1-score": 0.5,
    "support": 2199
  },
  "weighted avg": {
    "precision": 0.4961346066393815,
    "recall": 0.4961346066393815,
    "f1-score": 0.4961346066393815,
    "support": 2199
  }
}
```

**Pull request merged**